python/ray/includes/unique_ids.pxi

```diff
@@ -323,10 +328,40 @@ cdef class ClientActorRef(ActorID):
         if client.ray.is_connected() and not self.data.IsNil():
             client.ray.call_release(self.id)
 
+    def binary(self):
+        self._wait_for_id()
+        return self.data.Binary()
+
+    def hex(self):
+        self._wait_for_id()
+        return decode(self.data.Hex())
+
+    def is_nil(self):
+        self._wait_for_id()
+        return self.data.IsNil()
+
+    cdef size_t hash(self):
+        self._wait_for_id()
+        return self.data.Hash()
+
+    cdef CActorID native(self):
+        self._wait_for_id()
+        return <CActorID>self.data
+
     @property
     def id(self):
         return self.binary()
 
+    cdef _set_id(self, id):
+        check_id(id, CActorID.Size())
+        self.data = CActorID.FromBinary(<c_string>id)
+        client.ray.call_retain(id)
+
```

Comment:
Could you add some tests to ensure that the 'public API' for ClientObjects and non-client Objects are the same?